File: template/python/part1.py
from time import time


def part1(x):
    with open(x, "r") as f:
        return f.read().splitlines()


test_val = 0
if part1("test") == test_val:
    start_time = time()
    print('Solution is {}'.format(part1("input")))
    print("--- %s seconds ---" % (time.time() - start_time))
else:
    print("Test failed")

File: template/rust/src/part1.rs
use std::time::Instant;

fn solution(_data: String) -> u32 {
    0
}

pub fn print_solution(test_val: u32) {
    let start = Instant::now();

    let input = std::fs::read_to_string("./input").unwrap();
    let test = std::fs::read_to_string("./test").unwrap();

    if solution(test) != test_val {
        println!("Test failed");
    } else {
        println!("{}", solution(input));
        let duration = start.elapsed();
        println!("Elapsed time: {:.2?}", duration);
    }
}
